apidef/oas/operation_test.go

```diff
@@ -1113,3 +1165,295 @@ func verifyOASOperation(t *testing.T, op *openapi3.Operation, method string, cod
 		require.Equal(t, body, response.Content[contentType].Examples["default"].Value.Value)
 	}
 }
+
+func TestPathPartString(t *testing.T) {
+	tests := []struct {
+		name     string
+		pathPart pathPart
+		want     string
+	}{
+		{
+			name: "regular path part",
+			pathPart: pathPart{
+				name:    "users",
+				value:   "users",
+				isRegex: false,
+			},
+			want: "users",
+		},
+		{
+			name: "regex path part",
+			pathPart: pathPart{
+				name:    "customRegex1",
+				value:   ".*",
+				isRegex: true,
+			},
+			want: "{customRegex1}",
+		},
+		{
+			name: "named pattern format",
+			pathPart: pathPart{
+				name:    "id",
+				value:   "[0-9]+",
+				isRegex: true,
+			},
+			want: "{id}",
+		},
+		{
+			name: "named pattern with multiple colons",
+			pathPart: pathPart{
+				name:    "path",
+				value:   ".*:more:stuff",
+				isRegex: true,
+			},
+			want: "{path}",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.pathPart.String()
+			if got != tt.want {
+				t.Errorf("pathPart.String() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSplitPath(t *testing.T) {
+	tests := []struct {
+		name          string
+		path          string
+		expectedParts []pathPart
+		expectedRegex bool
+	}{
+		{
+			name: "standard path without regex",
+			path: "/users/documents/list",
+			expectedParts: []pathPart{
+				{name: "users", value: "users", isRegex: false},
+				{name: "documents", value: "documents", isRegex: false},
+				{name: "list", value: "list", isRegex: false},
+			},
+			expectedRegex: false,
+		},
+		{
+			name: "path with named parameter",
+			path: "/users/{id}/profile",
+			expectedParts: []pathPart{
+				{name: "users", value: "users", isRegex: false},
+				{name: "id", value: "", isRegex: true},
+				{name: "profile", value: "profile", isRegex: false},
+			},
+			expectedRegex: true,
+		},
+		{
+			name: "path with regex pattern",
+			path: "/users/{id:[0-9]+}/profile",
+			expectedParts: []pathPart{
+				{name: "users", value: "users", isRegex: false},
+				{name: "id", value: "[0-9]+", isRegex: true},
+				{name: "profile", value: "profile", isRegex: false},
+			},
+			expectedRegex: true,
+		},
+		{
+			name: "path with direct regex",
+			path: "/files/{.*}/download",
```

Comment:
this is invalid by mux; read through httputil/mux* ; i feel a significant part is already implemented in httputil, maybe needs extending (mux.go, we already have logic for `/*` and `/*/` which equals this patch matching case.

With `{.*}` - this mux tag is not a valid input, while `{_}` or `{_:.*}` would be. Is this a case we need to cover? Technically it would "work" in mux because `.*` would be treated as a `name`, using the default parameter regex. customRegex in this case should ignore {} and be isRegex=false

Reply:
@titpetric qq: [IsMuxTemplate](https://github.com/TykTechnologies/tyk/blob/83d4b12c7fb8bb7ea3a1caf2489cff4f4722a119/internal/httputil/mux.go#L73) is checking only for existence of curly braces which are opened and closed. And if it is returning true, whatever is inside `{}` is replaced with `([^/]+)`. 
This means `{id:[a-z]}` is also replaced with `([^/]+)`, is it expected?